File: src/BindOpen.Abstractions/Data/_Common/Enums/BdoExpressionKind.cs
namespace BindOpen.Data
{
    /// <summary>
    /// This enumeration represents the possible kinds for data expression.
    /// </summary>
    public enum BdoExpressionKind
    {
        /// <summary>
        /// Any.
        /// </summary>
        Any,

        /// <summary>
        /// Literal expression that can be script if it begins by "=".
        /// </summary>
        Auto,

        /// <summary>
        /// Literal expression is considered as default. A literal mode can then be applied.
        /// </summary>
        Literal,

        /// <summary>
        /// Script expression is considered as default.
        /// </summary>
        Script,

        /// <summary>
        /// Script word is considered as default.
        /// </summary>
        Word
    }

}

File: src/@kernel-dtos/src/data/enums/BdoExpressionKind.ts
/// <summary>
/// This enumeration represents the possible kinds of expression.
/// </summary>
export enum BdoExpressionKind
{
    /// <summary>
    /// Literal expression that can include scripts embedded in braces.
    /// </summary>
    Auto = "Auto",

    /// <summary>
    /// Literal expression is considered as default. A literal mode can then be applied.
    /// </summary>
    Literal = "Literal",

    /// <summary>
    /// Script expression is considered as default.
    /// </summary>
    Script = "Script",

    /// <summary>
    /// Word.
    /// </summary>
    Word = "Word",

    /// <summary>
    /// None. Such as script word.
    /// </summary>
    None = "None",

    /// <summary>
    /// Any.
    /// </summary>
    Any = "Any"
}
